tests/interactive/test_app.py

```diff
@@ -66,3 +69,27 @@ async def test_input():
 
         assert isinstance(app.current_module, ModuleOp)
         assert app.current_module.is_structurally_equivalent(expected_module)
+
+
+@pytest.mark.asyncio()
+async def test_buttons():
+    """Test pressing keys has the desired result."""
+    async with InputApp().run_test() as pilot:
+        app = cast(InputApp, pilot.app)
+
+        # Test clicking the "clear input" button
+        app.input_text_area.insert(
+            """
+        func.func @hello(%n : index) -> index {
+          %two = arith.constant 2 : index
+          %res = arith.muli %n, %two : index
+          func.return %res : index
+        }
+        """
+        )
+        await pilot.pause()
+        assert app.input_text_area.text != ""
```

Comment:
actually can you please test for the exact expected value?

Reply:
then we're gtg

Reply:
I'm not sure I get what the shared mutable state is?

Reply:
And how that's relevant to this != test

Reply:
Ah I see what happened, not sure why you call it shared mutable state, there's something I'm not getting still, but we can discuss that later.